provider/malicious_sources/resource_region_rule_block.go

```
@@ -140,38 +122,43 @@ func resourceRegionRuleBlockRead(d *schema.ResourceData, meta interface{}) error
 		d.SetId("")
```

the functions  like common.CallGetRuleDetailsFromRulesListUsingIdName are the main reasons for plugin crashed 

beacuse we are converting json response into map[string]interface{} without error handling suppose a graphql request succeded(status code =200) it shows you can not  create this rule ,our provider show it is crashed in logs not the graph ql does not succedded(means desired reponse is getting)


I think when we execute graphql call we have to implement it does not error field I think it should work

---

These errors are handled in respective provider code. Can you give me a reference

---

suppose for any reason either (server down or jwt does not have enough permission to create it) in this way graphql responds with status 200 but the response is like that  

{
    "errors": [
        {
            "message": "Validation error (FieldUndefined@[countrie]) : Field 'countrie' in type 'Query' is undefined",
            "locations": [
                {
                    "line": 4,
                    "column": 3
                }
            ],
            "extensions": {
                "classification": "ValidationError"
            }
        }
    ],
    "data": null
}

in recent code it only check the status of request so it never shows graphql exceptions but when you pass in the functions like above for extracting certain fields (by converting it in  map[string]interface{} ) it shows error that error is not handled and make the provider crashed in logs 

I hope I convey my point

---

Take Care this Error Handling in a another issue